src/views/Scopes/index.jsx

```diff
@@ -0,0 +1,37 @@
+import { hot } from 'react-hot-loader';
+import { Component } from 'react';
+import { Switch } from 'react-router-dom';
+import RouteWithProps from '../../components/RouteWithProps';
+import loadable from '../../utils/loadable';
+
+const ViewScopes = loadable(() =>
+  import(/* webpackChunkName: 'Clients.ViewScopes' */ './ViewScopes')
+);
+const ViewScope = loadable(() =>
+  import(/* webpackChunkName: 'Clients.ViewScope' */ './ViewScope')
+);
+
+@hot(module)
+export default class Scopes extends Component {
+  render() {
+    const {
+      match: { path },
+      ...props
+    } = this.props;
+
+    return (
+      <Switch>
+        <RouteWithProps
+          path={`${path}/:view(clients|roles)/:selectedScope/:selectedScopeView(clients|roles)?`}
```

Comment:
Am I correct in reading that the URL is `/auth/scopes/roles/:scope/roles`? Seems kinda redundant. What if we did something like: `/auth/clients/:scope` and `/auth/roles/:scope`? There may be another variation too.

Reply:
`/auth/clients/:scope` and `/auth/roles/:scope` are being used by the client and role views.

Reply:
The tab layout makes URLs ugly.

Reply:
We could skip putting the tabs in the URL, and just make the default tab show up.

Reply:
Works for me.